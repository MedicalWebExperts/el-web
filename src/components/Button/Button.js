import React from 'react';
import PropTypes from 'prop-types';
import theme from '../../styles';

const propTypes = {
  text: PropTypes.string.isRequired,
  outline: PropTypes.bool,
  color: PropTypes.oneOf(['primary', 'secondary', 'success', 'warning', 'danger']),
};

const defaultProps = {
  outline: false,
  color: 'primary',
};

const Button = ({ text, outline, color }) => {
  // check type
  let buttonClassName = 'default';
  if (outline) {
    buttonClassName = 'outline';
  }
  // check colors
  let backgroundColor = null;
  let textColor = null;
  switch (color) {
    case 'secondary':
      backgroundColor = theme.colors.secondary;
      textColor = theme.colors.buttonSecondaryText;
      break;
    case 'success':
      backgroundColor = theme.colors.success;
      textColor = theme.colors.buttonSuccessText;
      break;
    case 'warning':
      backgroundColor = theme.colors.warning;
      textColor = theme.colors.buttonWarningText;
      break;
    case 'danger':
      backgroundColor = theme.colors.danger;
      textColor = theme.colors.buttonDangerText;
      break;
    default:
      backgroundColor = theme.colors.primary;
      textColor = theme.colors.buttonPrimaryText;
      break;
  }

  return (
    <div>
      <style jsx>
        {`
          button {
            text-align: center;
            padding: 15px 30px;
            font-size: 16px;
            cursor: 'pointer';
          }
          button.default {
            background-color: ${backgroundColor};
            color: ${textColor};
            outline: none;
          }
          button.outline {
            background-color: ${theme.colors.transparent};
            border: solid 1px ${backgroundColor};
            color: ${backgroundColor};
            outline: none;
          }
        `}
      </style>
      <button type="button" className={buttonClassName}>
        {text}
      </button>
    </div>
  );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
