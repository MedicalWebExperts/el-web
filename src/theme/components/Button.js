export default {
  border: 'none',
  outline: 'none',
  textAlign: 'center',
  backgroundColor: '#ffffff',
  color: '#000000',
  padding: '10px 28px',
  fontSize: '14px',
  fontWeight: 'bold',
  cursor: 'pointer',
  borderRadius: '4px',
};
