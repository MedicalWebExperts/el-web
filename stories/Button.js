import React from 'react';
import { storiesOf } from '@storybook/react';

import { Button } from '../src';

const customStyle = {
  maxWidth: 200,
};

export default () => {
  storiesOf('Button', module).add('Default', () => (
    <Button text="Default" onClick={() => null} styles={customStyle} />
  ));
  storiesOf('Button', module).add('Secondary', () => (
    <Button text="Secondary" type="secondary" onClick={() => null} styles={customStyle} />
  ));
  storiesOf('Button', module).add('Success', () => (
    <Button text="Success" type="success" onClick={() => null} styles={customStyle} />
  ));
  storiesOf('Button', module).add('Warning', () => (
    <Button text="Warning" type="warning" onClick={() => null} styles={customStyle} />
  ));
  storiesOf('Button', module).add('Danger', () => (
    <Button text="Danger" type="danger" onClick={() => null} styles={customStyle} />
  ));
  storiesOf('Button', module).add('Outline', () => (
    <Button text="Outline" outline onClick={() => null} styles={customStyle} />
  ));
};
