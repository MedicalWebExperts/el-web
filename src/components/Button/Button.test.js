import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

const defaultProps = {
  text: 'Button Test',
};

const customProps = {
  text: 'Button Test',
  outline: true,
  type: 'success',
};

const customStyles = {
  color: '#ccc',
  marginTop: '20px',
};

describe('Component Button Snapshot', () => {
  it('using default values', () => {
    const tree = renderer.create(<Button {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('using other type button and style outline', () => {
    const tree = renderer.create(<Button {...customProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('added optional style inline prop', () => {
    const tree = renderer.create(<Button {...defaultProps} styles={customStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
