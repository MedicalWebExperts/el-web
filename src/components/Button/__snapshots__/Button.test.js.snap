// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button Snapshot added optional style inline prop 1`] = `
<div
  className="jsx-178200624"
>
  <button
    className="jsx-178200624 default"
    onClick={[Function]}
    style={
      Object {
        "color": "#ccc",
        "marginTop": "20px",
      }
    }
    type="button"
  >
    Button Test
  </button>
</div>
`;

exports[`Component Button Snapshot using default values 1`] = `
<div
  className="jsx-178200624"
>
  <button
    className="jsx-178200624 default"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    Button Test
  </button>
</div>
`;

exports[`Component Button Snapshot using other type button and style outline 1`] = `
<div
  className="jsx-3514065733"
>
  <button
    className="jsx-3514065733 outline"
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    Button Test
  </button>
</div>
`;
